skimage/color/colorlabel.py

```diff
@@ -195,34 +197,87 @@ def _label2rgb_overlay(label, image=None, colors=None, alpha=0.3,
     return result
 
 
-def _label2rgb_avg(label_field, image, bg_label=0, bg_color=(0, 0, 0)):
+def _apply_2d_func_to_image(func, image):
```

Comment:
I don't want to hardcode the fact that the image is 2D or 2D+c. This function makes sense for 3D images (such as 3D segmentations in microscopy).

Reply:
You could have `multichannel=False` as a keyword(-only) argument that determines whether to do apply `func` to the whole array or to the channels (last dim) separately. In that case, I think we already have such a function in `adapt_rgb`, although a better name for this would be `apply_per_channel`.

Reply:
That's an excellent point.

I believe `adapt_rgb` looks at the shape of the image to adapt a filter func. I create a function called `_apply_per_channel` per your suggestion.